src/odemis/driver/pwrmccdaq.py

```
@@ -364,6 +377,193 @@ def _update_power(self, value: List[float]):
+                raise ValueError(f"Axis '{axis_name}' has invalid 'do' channel {do_ch}")
+
+            self._axes_cfg[axis_name] = (do_ch, low, high, duration)
+            axes_def[axis_name] = model.Axis(unit=unit, choices={low, high})
+
+            if "di" in axis_cfg:
+                di_ch = axis_cfg["di"]
+                di_inverted = axis_cfg.get("di_inverted", False)
+                try:
+                    parent.channel_to_port(di_ch)
+                except ValueError:
+                    raise ValueError(f"Axis '{axis_name}' has invalid 'di' channel {di_ch}")
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)